services/web3/useCreateNewClaimableBatch.tsx

```diff
@@ -6,27 +6,33 @@ import { psyClaimsMainnet, psyClaimsSepolia } from "@/constants/contracts";
 import psyClaimsAbi from "@/abis/psyClaimsAbi.json";
 
 export const useCreateNewClaimableBatch = () => {
-  const { data, writeContract, isPending, error } = useWriteContract();
+  const { data, writeContract, isPending, error, reset } = useWriteContract();
 
   const { isLoading: isConfirming, isSuccess: isConfirmed } =
     useWaitForTransactionReceipt({
       hash: data
     });
 
   const createNewClaimableBatch = useCallback(
-    async (
-      merkleRoot: string,
-      deadline: string,
-      ipfsHash: string,
-    ) => {
-      return writeContract({
-        address: env.NEXT_PUBLIC_IS_MAINNET
-          ? psyClaimsMainnet
-          : psyClaimsSepolia,
-        functionName: "createNewClaimableBatch",
-        abi: env.NEXT_PUBLIC_IS_MAINNET ? psyClaimsAbi : psyClaimsAbi,
-        args: [merkleRoot, BigInt(deadline), ipfsHash],
-      });
+    async (merkleRoot: string, deadline: string, ipfsHash: string) => {
+      return writeContract(
+        {
+          address: env.NEXT_PUBLIC_IS_MAINNET
+            ? psyClaimsMainnet
+            : psyClaimsSepolia,
+          functionName: "createNewClaimableBatch",
+          abi: env.NEXT_PUBLIC_IS_MAINNET ? psyClaimsAbi : psyClaimsAbi,
+          args: [merkleRoot, BigInt(deadline), ipfsHash]
+        },
+        {
+          onSuccess() {
+            reset();
```

Comment:
this won't work well here. unfortunately, success will be true long before the transaction is completed and indexed